doc/kci_testsuite.md

```diff
@@ -0,0 +1,300 @@
+    extra_packages:
+      - git 
+      - autoconf 
+      - automake 
+      - gcc 
+      - make 
+      - pkg-config 
+      - e2fslibs-dev 
+      - libblkid-dev 
+      - zlib1g-dev 
+      - liblzo2-dev 
+      - asciidoc 
+      - xmlto 
+      - libzstd-dev
+      - python3-setuptools
+      - python3.7-dev
+    script: "scripts/buster-btrfs.sh"
+    test_overlay: "overlays/btrfs"
+```
+
+
+2. We will look into last three config entries namely, `extra_packages`, `script` and `test_overlay`.
+   `extra_packages` as name suggests it will add the listed packages from final rootfs image.
+
+3. Remember `script` path is relative to `kci_rootfs build --data-path` value. By default its `jenkins/debian/debos/script`
+   Here we need add new script named `buster-btrfs.sh`. Its a simple script to download the repo and build and install 
+   the btrfs-progs from source. 
+
+
+```
+$ cat jenkins/debian/debos/script/buster-btrfs.sh
+
+git clone https://github.com/kdave/btrfs-progs.git /btrfs-progs
+cd /btrfs-progs
+./autogen.sh && ./configure
+make
+make install
+```
+
+
+4. `test_overlay` option is used to create specific directory layout inside the rootfs image. In our case, 
+
+```
+$ tree jenkins/debian/debos/overlays/btrfs/
+jenkins/debian/debos/overlays/btrfs/
+└── usr
+    └── bin
+        └── btrfs-verify-cli.sh
+```
+
+   `/usr/bin/btrfs-verify-cli.sh` will be available on `buster-btrfs` rootfs image. These overlay scripts normally ran from LAVA job.
+
+```
+$ cat jenkins/debian/debos/overlays/btrfs/btrfs-verify-cli.sh
+
+cd  /btrfs-progs && TEST=001\* ./tests/cli-tests.sh
+```
+
+5. Make sure execute permission is applied on the scripts. 
+
+```
+chmod +x jenkins/debian/debos/overlays/btrfs/usr/bin/btrfs-verify-cli.sh
+chmod +x jenkins/debian/debos/script/buster-btrfs.sh
+```
+
+6. Now build `buster-btrfs` rootfs image using kci_rootfs. 
+
+7. Once rootfs image is created, you can use it with appropriate kernel image. Make sure that kernel config has 
+   appropriate btrfs configs and other file system specific entries enabled. 
+
+8. Add appropriate test entry with LAVA job definition for `btrfs-verify-cli.sh`. For example,
+
+```
+run:
+    steps:
+        - lava-test-case test-btrfs-cli-001 --shell ls /usr/bin/btrfs-verify-cli.sh
+
+```
+
+If you like read more about LAVA job definition, please check [here](https://docs.lavasoftware.org/lava/writing-tests.html)
+
+
+### How to add tests to LAVA
+
+#### Prerequisites
+
+- Rootfs image is built 
+- Rootfs image boots on a platform
+- Tests can be run
+
+If you're facing issues with tha above you may want to go through the section "How to add test suite to a rootfs image"
+
+#### Adding new test plan to LAVA
+
+1. Use the rootfs image to boot the device and produce a LAVA job definition for the tests.
+Upload rootfs image to kernelci-backend (storage)
+
+```
+kci_rootfs upload --rootfs-dir /path/to/rootf/image/ --upload-path server/path --api http://api.kernelci --token KCI_API_TOKEN
```

Comment:
Note: the `--api` and `--token` values can be put in the `kernelci.conf` config file rather than on the command line.

Also worth noting, most people won't have a token for `api.kernelci.org` (also the URL is broken in the example).  So maybe this can be replaced with something like `http://localhost:12345` with a comment to say the API needs to match the instance which the user has access to.

Reply:
Also, we can mention the staging workflow where people can submit a PR with a new rootfs definition which will get built on staging and then production kernelci.org once merged.